relay-server/src/actors/global_config.rs

```diff
@@ -313,14 +337,69 @@ mod tests {
         });
 
         Controller::start(Duration::from_secs(1));
-        let config = Arc::<Config>::default();
-        let service = GlobalConfigService::new(config.clone(), upstream).start();
+        let mut config = Config::default();
+        config.update_credentials();
+        let fetch_interval = config.global_config_fetch_interval();
+
+        let service = GlobalConfigService::new(Arc::new(config), upstream).start();
 
         assert!(service.send(Get).await.is_ok());
 
         Controller::shutdown(ShutdownMode::Immediate);
-        tokio::time::sleep(config.global_config_fetch_interval() * 2).await;
+        tokio::time::sleep(fetch_interval * 2).await;
 
         assert!(service.send(Get).await.is_ok());
     }
+
+    #[tokio::test]
+    #[should_panic]
+    async fn managed_relay_makes_upstream_request() {
+        relay_test::setup();
+        tokio::time::pause();
+
+        let (upstream, handle) = mock_service("upstream", (), |(), _| {
+            panic!();
```

Comment:
Instead of `#[should_panic]`, which is intended to be used for expected panics, we should rather make sure the upstream service returns a default config.

Reply:
IIUC, you suggest returning a response as if it were the actual upstream service. To actually return a global config we need to do an async operation in the closure, currently unstable in Rust. I think the `panic` makes the test easy to understand, and removes the complexity of dealing with `mock_service`. Maybe I'm missing something?

Reply:
I'm here with @jan-auer . The thing is with unit tests, it can also panic for completely different reason when something changes in the code, and therefore the tests will be testing completely wrong thing. 

It's better to make sure the behaviour is correct, but checking the status of some operation, like in this case that resulting config is `default`

Reply:
@TBS1996 and I spent some time on this today. In order to move forward with this PR, let's keep the `should_panic` for now. He'll link a follow-up issue here to clean this up. 

We'll implement some utilities to make mocking upstream requests easier. The boilerplate needed to achieve this now for a single test is too complex.